python/nav/report/generator.py

```diff
@@ -146,25 +150,24 @@ def parse_report(self, report_name):
         the access methods will probably fit here
         """
 
-        if self.config is None:
-            self.config = io.open(self.config_file, encoding='utf-8').read()
-            self.config_local = io.open(self.config_file_local, encoding='utf-8').read()
+        if not self.config_list:
+            self.config_list = [
+                io.open(config_file, encoding="utf-8").read()
+                for config_file in self.config_files
+            ]
         report_pattern = re.compile(
             r"^\s*" + report_name + r"\s*\{(.*?)\}$", re.M | re.S | re.I
         )
-        match = report_pattern.search(self.config)
-        local_match = report_pattern.search(self.config_local)
+        matches = (report_pattern.search(config) for config in self.config_list)
+        matches = [m for m in matches if m]
```

Comment:
Hm, hm: if we only run on python 3.8 or newer, this is a good use-case for the walrus-operator:

```python
matches = (match for config in self.config_list if (match := report_pattern.search(config))
```

Reply:
alas, no walrus for us quite yet...